include/aws/mqtt/client.h

```diff
@@ -63,6 +63,9 @@ typedef void(aws_mqtt_client_on_connection_interrupted_fn)(
     int error_code,
     void *userdata);
 
+/* Called if the connection to the server is closed by user request */
+typedef void(aws_mqtt_client_on_connection_closed_fn)(struct aws_mqtt_client_connection *connection, void *userdata);
```

Comment:
Do we want to add a struct to this callback for future possible extension? We currently use an empty struct for that possibility in MQTT5 in callbacks like lifecycle events stop and attempting connect.